hal/src/nRF52840/ble_hal.cpp

```diff
@@ -2035,22 +2033,22 @@ int BleObject::ConnectionsManager::rejectPairing(hal_ble_conn_handle_t connHandl
     CHECK_TRUE(connection, SYSTEM_ERROR_NOT_FOUND);
     switch (connection->pairState) {
         case BLE_PAIRING_STATE_SEC_REQ_RECEIVED: {
-            connection->pairState = BLE_PAIRING_STATE_REJECTED;
             int ret = sd_ble_gap_authenticate(connection->info.conn_handle, nullptr);
             CHECK_NRF_RETURN(ret, nrf_system_error(ret));
+            connection->pairState = BLE_PAIRING_STATE_REJECTED;
```

Comment:
Can you add a comment on why we are no longer ignoring errors for setting the internal pairing state to rejected?

Reply:
Will add: 
>We should not tamper the pairing state to be out of sync with the SoftDevice. Only if the SD call returns success we then mark the pairing state as rejected.

Reply:
:+1: